### DISCUSSION TITLE
For best practices in dynamic rescaling

### DISCUSSION THREAD

**Discussion by nagakagachi** [jrouwe/JoltPhysics#700] [Q&A] [answered]
Hi!
I am looking for a recommended way to dynamically change the scale of an object in Jolt.
I'm thinking perhaps replacing it with a shape that has been rescaled, but I'd like to know if there is a proper way to do this.
Thanks.

naga.

**Comment by jrouwe** [comment, marked as answer]
Hello,

The recommended way would be to create a new ```ScaledShape``` and replace the existing ```ScaledShape``` using ```BodyInterface::SetShape```. There is no ```SetScale``` function because it would introduce a race condition when doing multithreaded queries / modifications.

Beware though that changing scales (or shapes in general) makes it easy for bodies to be pushed through the floor. Consider for example a hollow tube with a marble in it that 'just fits', if you enlarge the marble it will be pushed out in a random direction (or at least heavily jitter) and that random direction may be not a direction you desire and there's no way to control that.

**Reply by nagakagachi** [reply]
I was able to achieve the desired behavior by replacing the new shape by BodyInterface::SetShape.

In my case, I can tolerate unnatural motion due to scaling changes, so this method works fine.
Thanks!
